### DISCUSSION TITLE
Why was experimental_standaloneMiddleware deprecated?

### DISCUSSION THREAD

**Discussion by SiNONiMiTY** [trpc/trpc#5732] [General]
The use case is extremely simpler than the one advertised by the document `.concat()`

Here is an example

I want to create a procedure in which I can alter the context in a simple manner, chainable like this as I depend on the provided .input()

```
t.procedure
  .input(combinedInputThatSatisfiesBothMiddlewares)
  .use(valueAUppercaserMiddleware)
  .use(valueBUppercaserMiddleware)
```

I am having a hard time achieving this on .concat() whereas experimental_standaloneMiddleware() perfectly fits this use case.

Or maybe I am missing something?

@KATT can you please provide your insights as to why this feature was deprecated. Thank you for your hard work :)

**Comment by do4Mother** [comment]
is there any update for this? i have the same use case with you that `experimental_standaloneMiddleware()` is perfect instead of using `.concat()`

**Comment by hugo082** [comment]
Any update here? I don't understand either

**Reply by Nick-Lucas** [reply]
It's a quite off in the past so my memory is a bit leaky here, but standalone middlewares:

1. had some type inference issues which proved hard to resolve the way they were written
2. were one of several attempts we made at solving a similar problem

[concat](https://trpc.io/docs/server/middlewares#concat) by contrast solved a lot of the underlying issues, allowing you to define required input/context/meta types in a way that tRPC could later merge, and create more complex middleware chains out the box. That's why it's recommended today, and we've deprecated other attempts in favour of a single canonical approach to re-usable middlewares
